src/components/Checkbox/Checkbox.tsx

```diff
@@ -29,6 +29,14 @@ export const Checkbox = ({
 }: CheckboxProps) => {
   const randomId = useId();
   const inputId = checkboxId || 'checkbox-' + randomId;
+  const [usingTab, setUsingTab] = useState<boolean>(false);
+
+  document.body.addEventListener('mousedown', () => setUsingTab(false));
+
+  document.body.addEventListener(
+    'keydown',
+    (event) => event.key === 'Tab' && setUsingTab(true),
+  );
```

Comment:
This shoul be added in a `useEffect`, and the listener should be cleaned up when the component unmounts/rerenders. Otherwise, there will be a new listener created whenever the component rerenders, and never getting cleaned up.

Wrote this by hand, so there are probably some errors 😅

```suggestion
  useEffect(() => {
    const mouseDownListener = () => setUsingTab(false);
    const tabKeyListener = () => (event) => event.key === 'Tab' && setUsingTab(true);
    document.body.addEventListener('mousedown', mouseDownListener);
    document.body.addEventListener('keydown', tabKeyListener);
    
    return (() => {
    	document.body.removeEventListener('mousedown', mouseDownListener);
    	document.body.removeEventListener('keydown', tabKeyListener);
    });
  });
```

That said, I'm not sure we should do anything like this at all.

I assume the goal is to not show the outline when clicking with mouse, but do show it when tabbing?

If that is the goal, I propose to change the css instead to something like this:

```css

// fallback for browsers who do not yet support :has(:focus-visible)
// will show outline on mouse click as well
@supports not selector(:has(:focus-visible)) {
  .wrapper:not(.wrapper--disabled):focus-within {
    outline: 2px solid var(--interactive_components-colors-focus_outline);
    outline-offset: 2px;
  }
}


// for browsers who do support :has(:focus-visible), outline will only be shown on keyboard navigation
.wrapper:not(.wrapper--disabled):has(:focus-visible) {
  outline: 2px solid var(--interactive_components-colors-focus_outline);
  outline-offset: 2px;
}

```


This solves both goals nicely, and we dont need to complicate the code with JS.
When Firefox adds support for it, it will also get the "prettier" variant, where the outline only occurs on keyboard navigation.

Reply:
Good catch, somehow I didn't think about this rerendering problem🤦 However I also like the solution with the `:has` selector better, so I have implemented that now.